### PR TITLE
Updated menu example to be more accessible and realistic

### PR DESCRIPTION
**Pull request by PK106552**
**What was changed:**
Updated example to be more accessible and realistic.

**Why it was changed:**
Example were confusing and unrealistic.

**Additional Details**
This PR resolves:

[UXPLATFORM-8361](https://jira2.cerner.com/browse/UXPLATFORM-8361)

### CODE REVIEW

File: packages/terra-menu/CHANGELOG.md [hunk 1]
```diff
@@ -2,6 +2,9 @@
 
 ## Unreleased
 
+* Changed
```
Comment (supreethmr) [inline packages/terra-menu/CHANGELOG.md:5]: this `changelog` is not required as we are not making any changes on terra-menu component.

**Review by supreethmr — APPROVED** [review]
+1 with above comments addressed.

**Review by sdadn — APPROVED** [review]
Approved, provided @supreethmr 's comments are addressed.

File: packages/terra-framework-docs/src/terra-dev-site/doc/menu/example/BasicMenu.jsx [hunk 2]
```diff
@@ -53,7 +54,7 @@ class BasicMenu extends React.Component {
 
   handleToggle1OnClick() {
     this.setState(prevState => ({ toggle1Selected: !prevState.toggle1Selected }));
-    this.handleRequestClose();
+    // this.handleRequestClose();
```
[the remark this answers is not in the document]
Comment (PK106552) [reply packages/terra-framework-docs/src/terra-dev-site/doc/menu/example/BasicMenu.jsx]: Updated

File: packages/terra-menu/CHANGELOG.md (hunk 1 above)
Comment (PK106552) [reply packages/terra-menu/CHANGELOG.md]: Removed

**Pull request merged**